freedesktop.org.xml.in

```diff
@@ -6316,4 +6320,15 @@ command to generate the output files.
     <glob pattern="*.stl"/>
   </mime-type>
 
+<mime-type type="application/x-gcode">
+    <_comment>G-code file</_comment>
+    <sub-class-of type="text/plain"/>
+    <generic-icon name="text-x-generic"/>
+    <magic priority="10">
+      <match type="string" value="; " offset="0"/>
+      <match type="string" value="( " offset="0"/>
```

Comment:
Aah. Never seen that in RepRap related GCODE.

What about adding `G` for files starting without comments? I found an old Skeinforge generated Gcode on my drive and it doesn't have any:

```
G90
G21
G28                
G92 E0
G1 E2 F1200
G1 E1 F1200
G92 E0
G1 X5.0 Z0.3 F200
G1 Y75.0 E2.0  F1000.0
G1 Y100.0 E2.3  F1000.0
G90
G21
M103
M105
M113 S1.0
M108 S12.86
G1 X60.844659 Y90.0 Z0.3 F7800.0
G1 F2040.0
G1 E1.1
G1 F7800.0
M101
G1 X61.749443 Y98.292453 Z0.3 F840.0 E1.2842446
...
```